src/yarp.c

```diff
@@ -9773,6 +9773,12 @@ parse_arguments_list(yp_parser_t *parser, yp_arguments_t *arguments, bool accept
         // argument to this method call.
         parse_arguments(parser, arguments, true, YP_TOKEN_EOF);
 
+        // If we have done with the arguments and still not consumed the comma, then we have a trailing comma
+        // where we need to check whether it is allowed or not.
+        if (parser->previous.type == YP_TOKEN_COMMA && !match1(parser, YP_TOKEN_SEMICOLON)) {
+            yp_diagnostic_list_append(&parser->error_list, parser->previous.start, parser->previous.end, YP_ERR_EXPECT_ARGUMENT);
+        }
```

Comment:
```suggestion
        // If we have done with the arguments and still not consumed the comma,
        // then we have a trailing comma where we need to check whether it is
        // allowed or not.
        if (parser->previous.type == PM_TOKEN_COMMA && !match1(parser, PM_TOKEN_SEMICOLON)) {
            pm_parser_err_previous(parser, PM_ERR_EXPECT_ARGUMENT);
        }
```

Reply:
Sorry this took so long, but yeah this looks good. I've added a suggestion here for how to update it since it has been so long.